### PR TITLE
[SPARK-34305][SQL][TESTS] Unify v1 and v2 ALTER TABLE .. SET SERDE tests

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?

- Move parser tests from DDLParserSuite to AlterTableSetSerdeParserSuite.
- Port DS v1 tests from DDLSuite and other test suites to v1.AlterTableSetSerdeSuite.
- Add a test for DSv2 ALTER TABLE .. SET SERDE to v2.AlterTableSetSerdeSuite.

### Why are the changes needed?
To improve test coverage.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
By running new test suites:

> $ build/sbt -Phive-2.3 -Phive-thriftserver "test:testOnly *AlterTableSetSerdeSuite"
> $ build/sbt -Phive-2.3 -Phive-thriftserver "test:testOnly *AlterTableSetSerdeParserSuite"

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/AlterTableSetSerdeParserSuite.scala [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    val sql = "ALTER TABLE my_tab SET SERDE 'serde' " +
+      "WITH SERDEPROPERTIES('key_without_value', 'key_with_value'='x')"
+    val errMsg = intercept[ParseException] {
+      parsePlan(sql)
+    }.getMessage
+    assert(errMsg.contains("Operation not allowed"))
+    assert(errMsg.contains("key_without_value"))
+  }
+
+  test("alter table: SerDe properties") {
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/AlterTableSetSerdeParserSuite.scala:38]: Could you be more consistent and use either `-` or `:`

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/command/AlterTableSetSerdeSuite.scala [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+class AlterTableSetSerdeSuite extends v1.AlterTableSetSerdeSuiteBase with CommandSuiteBase {
+
+  override def afterEach(): Unit = {
+    try {
+      // drop all databases, tables and functions after each test
+      spark.sessionState.catalog.reset()
+    } finally {
+      super.afterEach()
+    }
+  }
```
Comment (MaxGekk) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/command/AlterTableSetSerdeSuite.scala:39]: Why do you need this? Can't you make each test self-contained and create/delete all dependencies inside the test as we do in other test suites?

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/AlterTableSetSerdeParserSuite.scala [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+      """.stripMargin
+    val parsed4 = parsePlan(sql4)
+    val expected4 = SetTableSerDeProperties(
+      UnresolvedTable(Seq("table_name"), "ALTER TABLE ... SET [SERDE|SERDEPROPERTIES]", hint),
+      Some("org.apache.class"),
+      Some(Map("columns" -> "foo,bar", "field.delim" -> ",")),
+      Some(Map("test" -> "1", "dt" -> "2008-08-08", "country" -> "us")))
+    comparePlans(parsed4, expected4)
+
+    val sql5 =
```
[the remark this answers is not in the document]
Comment (panbingkun) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/command/AlterTableSetSerdeParserSuite.scala:89]: I will do it

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/v1/AlterTableSetSerdeSuite.scala [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+        assert(isDatasourceTable, "InMemoryCatalog only supports data source tables")
+      }
+      sql(s"CREATE TABLE $t (col1 int, col2 string, a int, b int) $defaultUsing " +
+        s"PARTITIONED BY (a, b)")
+
+      val catalog = spark.sessionState.catalog
+      val tableIdent = TableIdentifier("tbl", Some("ns"))
+      def checkSerdeProps(expectedSerdeProps: Map[String, String]): Unit = {
+        val serdeProp = catalog.getTableMetadata(tableIdent).storage.properties
+        if (isUsingHiveMetastore) {
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/v1/AlterTableSetSerdeSuite.scala:62]: Could you extract common code to functions in `AlterTableSetSerdeSuiteBase`, and create dedicated tests for Hive and In-Memory catalogs. Please, invoke the common code from catalog specific tests. This is common convention in the unified tests for v1 and v2 catalogs.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/v1/AlterTableSetSerdeSuite.scala (hunk 4 above)
Comment (panbingkun) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/command/v1/AlterTableSetSerdeSuite.scala:62]: OK，I will do it.

Comment (panbingkun) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/command/v1/AlterTableSetSerdeSuite.scala]: Done

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @panbingkun.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala [hunk 5]
```diff
@@ -227,23 +226,21 @@ class InMemoryCatalogedDDLSuite extends DDLSuite with SharedSparkSession {
...
-
-  protected val reversedProperties = Seq(PROP_OWNER)
+trait DDLSuiteBase extends SQLTestUtils {
 
   protected def isUsingHiveMetastore: Boolean = {
     spark.sparkContext.conf.get(CATALOG_IMPLEMENTATION) == "hive"
   }
 
   protected def generateTable(
-      catalog: SessionCatalog,
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala:239]: the previous 4 spaces indentation was correct

**Pull request closed without merging**